floyd/client/files.py

```diff
@@ -148,6 +149,13 @@ def create_tarfile(source_dir, filename="/tmp/contents.tar.gz"):
                             exit_msg=("Permission denied. Make sure to have read permission "
                                       "for all the files and directories in the path: %s")
                             % (source_dir))
+        # OSError: [Errno 28] No Space Left on Device (IOError on python2.7)
+        elif e.errno == errno.ENOSPC:
+            warn_purge_exit(info_msg="No space left. Removing compressed data...",
+                            filename=filename,
+                            exit_msg=("No space left when compressing your data in: %s.\n"
+                                      "Make sure to have enough space before uploading your data.")
+                            % (tempfile.mkdtemp()))
```

Comment:
it's better to get the directory from `filename` than using `tempfile.mkdtemp()`